FW-73 Cognito Integrated Auth

plasticviking opened the pull request:
# Changelog
 - Introduce a new bundle, `FirstVoicesCognitoAuth` that hooks into Nuxeo's `UserManager` to allow authentication directly against Cognito via the AWS Java SDK
 - Disabled by default, enable by setting `fv.cognito.authenticateWithCognito` to `true` in Nuxeo configuration. 5 other parameters must also be set for this to work -- see the updated `nuxeo.defaults`.
 - Works in conjunction with the AWS Cognito Lambda Migration trigger to automatically migrate users
 - Requires that the user exists in the Nuxeo directory (that is, merely having a user in Cognito is not enough, there must be a matching FirstVoices user).

davidlacho commented:
@plasticviking we began strict checkstyle enforcement and he build is failing on a few classes:

```2020-05-21T12:55:23.9800398Z [INFO] --- maven-checkstyle-plugin:3.1.1:check (default) @ FirstVoicesCognitoAuth ---
2020-05-21T12:55:23.9801389Z [INFO] There are 6 errors reported by Checkstyle 8.29 with https://raw.githubusercontent.com/First-Peoples-Cultural-Council/fv-checkstyle/master/java/java-fv-google-style-strict.xml ruleset.
2020-05-21T12:55:23.9802039Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAwareUserManagerConfigurationService.java:[10] (sizes) LineLength: Line is longer than 100 characters (found 101).
2020-05-21T12:55:23.9802685Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAuthenticationService.java:[31] (javadoc) JavadocTagContinuationIndentation: Line continuation have incorrect indentation level, expected level should be 4.
2020-05-21T12:55:23.9803352Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAuthenticationService.java:[32] (javadoc) JavadocTagContinuationIndentation: Line continuation have incorrect indentation level, expected level should be 4.
2020-05-21T12:55:23.9804335Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAuthenticationServiceImpl.java:[10] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - com.amazonaws.services.cognitoidp.model.*.
2020-05-21T12:55:23.9805405Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAuthenticationServiceImpl.java:[10] (imports) CustomImportOrder: Wrong lexicographical order for 'com.amazonaws.services.cognitoidp.model.*' import. Should be before 'java.util.Map'.
2020-05-21T12:55:23.9805915Z [ERROR] src/main/resources/seam.properties:[1] (misc) NewlineAtEndOfFile: File does not end with a newline.```

plasticviking commented:
> @plasticviking we began strict checkstyle enforcement and the build is failing on a few classes:
> 
> ```
> 2020-05-21T12:55:23.9801389Z [INFO] There are 6 errors reported by Checkstyle 8.29 with https://raw.githubusercontent.com/First-Peoples-Cultural-Council/fv-checkstyle/master/java/java-fv-google-style-strict.xml ruleset.
> 2020-05-21T12:55:23.9802039Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAwareUserManagerConfigurationService.java:[10] (sizes) LineLength: Line is longer than 100 characters (found 101).
> 2020-05-21T12:55:23.9802685Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAuthenticationService.java:[31] (javadoc) JavadocTagContinuationIndentation: Line continuation have incorrect indentation level, expected level should be 4.
> 2020-05-21T12:55:23.9803352Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAuthenticationService.java:[32] (javadoc) JavadocTagContinuationIndentation: Line continuation have incorrect indentation level, expected level should be 4.
> 2020-05-21T12:55:23.9804335Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAuthenticationServiceImpl.java:[10] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - com.amazonaws.services.cognitoidp.model.*.
> 2020-05-21T12:55:23.9805405Z [ERROR] src/main/java/ca/firstvoices/cognito/AWSAuthenticationServiceImpl.java:[10] (imports) CustomImportOrder: Wrong lexicographical order for 'com.amazonaws.services.cognitoidp.model.*' import. Should be before 'java.util.Map'.
> 2020-05-21T12:55:23.9805915Z [ERROR] src/main/resources/seam.properties:[1] (misc) NewlineAtEndOfFile: File does not end with a newline.```
> ```

Thanks @davidlacho I've fixed these and added checkstyle as a precommit hook on my local branch.

Looks like my IDE has some issues with formatting that require manual intervention -- somewhat frustrating. https://youtrack.jetbrains.com/issue/IDEA-121723

davidlacho commented:
@plasticviking Awesome! That's unfortunate. If there are any checkstyle rules that are very frustrating and please submit a PR to the checkstyle repo to remove them. We're still fine tuning our checkstyle rules.

dy2288 commented:
Any thoughts on why the build is failing in Github actions @davidlacho @plasticviking?

plasticviking commented:
> Any thoughts on why the build is failing in Github actions @davidlacho @plasticviking?

It seems to be failing in checking dependencies for the frontend. There are no frontend changes in the code, but I noticed the frontend failing to build on my local machine. I temporarily had upgraded `babel/core` and `babel/preset-env` to get the frontend to build successfully on my local machine (none of these changes are present in the committed code, but I found it was necessary to fix the build locally so perhaps it's necessary in the CI environment also).

davidlacho commented:
On Friday GitHub was throwing several 500 errors on their API and I'm guessing this affected fetching node modules. It seems to be passing the frontend build, and now the issue seems to be that we updated to 3.4.3-SNAPSHOT. Committed directly to this branch to see if that resolves issue.

davidlacho commented:
@plasticviking while you're working on the forgotten password flow, do you mind updating the package version to match master? That's what the build is failing on. Thanks!

plasticviking commented:
> @plasticviking while you're working on the forgotten password flow, do you mind updating the package version to match master? That's what the build is failing on. Thanks!

Hi David. This new version will correctly reset the Cognito password to match, and has the package version number updated to match master.
